cmd/diff.go

```diff
@@ -1,44 +1,76 @@
 package cmd
 
 import (
+	"errors"
 	"fmt"
-	"github.com/spf13/cobra"
+	"path/filepath"
 	"time"
+
+	"github.com/spf13/cobra"
 )
 
 var timeString string
 
+func init() {
+	diffCmd.Flags().UintVarP(&cfg.NumWorkers, "workers", "w", 2, "number of concurrent file-copy workers")
+	diffCmd.Flags().StringVarP(&timeString, "time", "t", "", "reference time with format: 20060102T150405")
+	rootCmd.AddCommand(diffCmd)
+}
+
 var diffCmd = &cobra.Command{
 	Use:   "diff",
 	Short: "differential backup",
 	Long:  "with differential backup only new or modified files and directories are being backed-up",
 	Args: func(cmd *cobra.Command, args []string) error {
 		if len(args) != 2 {
-			return fmt.Errorf("arguments mismatch, expecting 2 arguments")
+			return errors.New("arguments mismatch, expecting 2 arguments")
 		}
 		cfg.Src = args[0]
 		cfg.Target = args[1]
 
-		assertedTime, err := time.Parse(timeFormat, timeString)
+		if timeString == "" {
+			return errors.New("time string cannot be empty")
+		}
+		assertedTime, err := parseTime(timeString)
 		if err != nil {
-			return fmt.Errorf("failed to parse time flag value")
+			return fmt.Errorf("failed to parse time flag value: %v", err)
 		}
 		if !assertedTime.Before(time.Now()) {
-			return fmt.Errorf("reference time flag value is in the future")
+			return errors.New("reference time flag value is in the future")
 		}
-		cfg.RecoveryReferenceTime = assertedTime
+		cfg.ReferenceTime = assertedTime
 
 		return nil
 	},
-	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Printf("numWorkers: %v\n", cfg.NumWorkers)
-		fmt.Printf("src: %v\n", cfg.Src)
-		fmt.Printf("target: %v\n", cfg.Target)
-	},
-}
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// list
+		listDirPath = filepath.Join(rootDirPath, listDirName)
+		if err := ltCmd.RunE(cmd, args); err != nil {
+			return err
+		}
 
-func init() {
-	diffCmd.Flags().UintVarP(&cfg.NumWorkers, "workers", "w", 2, "number of concurrent file-copy workers")
-	diffCmd.Flags().StringVarP(&timeString, "time", "t", "", "reference time with format: 2006-01-02T15:04:05")
-	rootCmd.AddCommand(diffCmd)
+		// skeleton
+		skeletonWorkDir = filepath.Join(rootDirPath, dirSkeletonDirName)
+		dirsListFilePath = listDirsPath
+		if err := skeletonCmd.RunE(cmd, args); err != nil {
```

Comment:
Let's run the skeleton command here with the report option

Reply:
The default value for the skeleton validation mode is already `report`:
```go
skeletonCmd.Flags().StringVarP(&validationMode, "dir-validation-mode", "v", rberrors.Report, "validation mode for directories short list (none, report, block)")
```